app/ml_models/rnn/data_tools.py

```diff
@@ -1,34 +1,46 @@
-import torch
 import os
+from typing import List
+
+import torch
 from torch.utils.data import Dataset
 
+from app.ml_models.rnn.vocabulary import Vocabulary
+
 
 class WordLevelDataset(Dataset):
     def __init__(
         self,
         prefix: str = "../../../data/",
-        filename_dataset: str = "straattaal.txt",
+        filename_datasets: List[str] = ["straattaal.txt"],
+        vocabulary: Vocabulary = None,
         filename_vocab: str = "vocabulary.txt",
     ):
-        """Initialize a WordLevelDataset object.
+        """
+        Initialize a WordLevelDataset object and accompanying Vocabulary object if necessary.
 
         Args:
             prefix: The prefix to the folder containing the dataset.
-            filename_dataset: Full filename of dataset to be appended to the prefix.
-            filename_vocab: Full filename of vocabulary to be appended to the prefix.
+            filename_datasets: A list of full filenames of datasets to be appended to the prefix.
+            vocabulary: Optional Vocabulary object (see vocabulary.py). If set to None, vocabulary will be loaded from filename_vocab.
+            filename_vocab: Optional, only necessary if the vocabulary argument is None. Full filename of vocabulary to be appended to the prefix.
+
```

Comment:
nit
```suggestion
            vocabulary: Optional Vocabulary object (see vocabulary.py). If set to None, vocabulary will be loaded from filename_vocab.
            filename_vocab: Optional, only necessary if the vocabulary argument is None. Full filename of vocabulary to be appended to the prefix.
```

Reply:
Wait does it remove the empty line in my suggestion?

Reply:
I will commit anyway, to give you a green block